app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -0,0 +1,91 @@
+package com.github.se.cyrcle.model.parking
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel for the Parking feature.
+ *
+ * @param imageRepository the repository for the Parking feature
+ */
+class ParkingViewModel(
+    private val imageRepository: ImageRepository,
+    private val parkingRepository: ParkingRepository
+) : ViewModel() {
+
+  // a state that holds all displayed parkings slots /selected parkings
+  private val _selectedParkings = MutableStateFlow<List<Parking>>(emptyList())
```

Comment:
I think another name may be appropriate for this. Selected makes it sound like the user actively selects them, when it doesn't seem that that's always the case from the above comment. Maybe processedParkings would be better?